### ISSUE TITLE
Rename palette.styl

### ISSUE THREAD

**Issue by newsbielt703** [vuejs/vuepress#2108]
## Feature request

`palette.style` should be renamed since it's not only relevant to colors.

See [config.styl](https://github.com/vuejs/vuepress/blob/master/packages/%40vuepress/core/lib/client/style/config.styl).

#### What problem does this feature solve?

The name may confuse developers.

#### What does the proposed API look like?

rename  `palette.style` to `variables .style` or something else.

#### Are you willing to work on this yourself?
Yes

**Comment by meteorlxy** [comment]
Breaking change

**Comment by meteorlxy** [comment]
To avoid a major version release, we can keep the `palette.styl` and display a deprecation notice if `palette.styl` exists.

**Comment by newsbielt703** [comment]
Sounds good.👌

**Comment by Mister-Hope** [comment]
> To avoid a major version release, we can keep the `palette.styl` and display a deprecation notice if `palette.styl` exists, and totally remove it in next major version.

Yes, we can support both can give a warning in the console screen. So when can we have this feature?

**Comment by newsbielt703** [comment]
@Mister-Hope I'm a bit busy these days. If you'd like to take it, go ahead ~

**Comment by Mister-Hope** [comment]
> @Mister-Hope I'm a bit busy these days. If you'd like to take it, go ahead ~

But a rfc here, what shall we rename it? variables? I agreed with this one, but a better file name cound be better. Any ideas with a shorter one?

**Comment by JoaquinDecima** [comment]
I generate https://github.com/vuejs/vuepress/pull/2943 to fix this issues
